proxy/src/main/java/com/wavefront/agent/preprocessor/ReportPointObfuscateTransformer.java

```diff
@@ -60,29 +60,36 @@ public ReportPoint apply(@Nullable ReportPoint reportPoint) {
           String metric = reportPoint.getMetric();
           metric = encode(metric);
           reportPoint.setMetric(metric);
+          ruleMetrics.incrementRuleAppliedCounter();
           break;
+
         case "sourceName":
+        case "hostName":
```

Comment:
It seems like based on existing rules sourceName is where this logic fits in. E.g. from docs. Also once sourceName is there used hostName does not have relevance. 
https://docs.wavefront.com/proxies_preprocessor_rules.html#point-altering-rules. We also need. to support matches parameter. Please look at processors like SpanBlockFilter
example:case "sourceName":
          if (compiledPattern.matcher(span.getSource()).matches()) {
            ruleMetrics.incrementRuleAppliedCounter();
            return false;